spec/index.md

```diff
@@ -1954,6 +1954,15 @@ The optional `settings` parameter can be used to set the following settings:
 
     Default value: 5_000_000_000_000 (5 trillion cycles).
 
+-   `wasm_heap_memory_limit` (`nat`)
+
+    Must be a number between 0 and 2<sup>32</sup>-1, inclusively, and indicates the upper limit on the WASM heap memory consumption of the canister.
+
+    An operation (update method, global timer, heartbeat, canister init, canister post_upgrade) that causes the WASM heap memory consumption to exceed this limit will trap.
+    The WASM heap memory limit is ignored for query methods, response callback handlers, and canister pre_upgrade.
+
+    Default value: 4_294_967_296 (4 GiB).
```

Comment:
It seems in line with reserved_cycles_limit to always have a limit and just set it to a high value (e.g., 4GiB) to effectively disable the limit.

Reply:
Let's set the default value to 3GiB to keep the spec simple.

The actual rollout will be more smart about it:
- if the canister memory usage is below 2.5GiB, then we set the value to 3GiB.
- otherwise, we set the value to min(current_memory + 0.5GiB, 4GiB).
^ Does it make sense?

Reply:
Yeah, that makes sense. I didn't specify it though since it's a one-off action.